File: backend/src/main/java/com/gxs/bank/repository/ContactRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.ContactSubmission;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import java.util.List;
import java.util.UUID;

@Repository
public interface ContactRepository extends JpaRepository<ContactSubmission, UUID> {
    List<ContactSubmission> findByEmailOrderByCreatedAtDesc(String email);
    List<ContactSubmission> findAllByOrderByCreatedAtDesc();
}

File: backend-python/com/gxs/bank/repository/ContactRepository.py
from com.gxs.bank.model.ContactSubmission import ContactSubmission
from com.gxs.bank.repository._base import BaseRepository


class ContactRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, ContactSubmission)

    def findByEmailOrderByCreatedAtDesc(self, email: str):
        return (
            self.db.query(ContactSubmission)
            .filter(ContactSubmission.email == email)
            .order_by(ContactSubmission.createdAt.desc())
            .all()
        )

    def findAllByOrderByCreatedAtDesc(self):
        return self.db.query(ContactSubmission).order_by(ContactSubmission.createdAt.desc()).all()
